search/includes/classes/class-search.php

```diff
@@ -237,4 +240,44 @@ static function ep_skip_query_integration( $skip ) {
 		// The filter is checking if we should _skip_ query integration
 		return ! ( $query_integration_enabled || $query_integration_enabled_legacy );
 	}
+
+	/**
+	 * Filter for ep_pre_request_host
+	 *
+	 * Return the next host in our enpoint list if it's defined. Otherwise, return the last host.
+	 */
+	static function filter__ep_pre_request_host( $host, $failures, $path, $args ) {
```

Comment:
This shouldn't actually be `static` (since we call it on the `$this` instance from the `add_filter()` call).